x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -745,30 +749,49 @@ class AgentPolicyService {
           }),
     };
 
+    const permissions = (await storedPackagePoliciesToAgentPermissions(
+      soClient,
+      agentPolicy.package_policies
+    )) || { _fallback: DEFAULT_PERMISSIONS };
+
+    permissions._elastic_agent_checks = {
+      cluster: DEFAULT_PERMISSIONS.cluster,
+    };
+
+    // TODO fetch this from the elastic agent package
```

Comment:
Could you file an issue for this and link https://github.com/elastic/integrations/issues/953 as a blocker?